### PR TITLE
feat: add dropzone with lightbox preview

### PR DESCRIPTION
**Pull request by rizqitsani**
- Yang di recap belum dikerjain, kayanya bisa dibuat image gallery gitu aja, cuma belum ada filenya
- Type StepTwoData bisa dibuat FileWithPath[] cuma yup jadi error

### CODE REVIEW

File: components/Forms/DropzoneInput.tsx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        <div className='flex items-center flex-1 w-0'>
+          <HiOutlinePaperClip
+            className='flex-shrink-0 w-5 h-5 text-gray-400'
+            aria-hidden='true'
+          />
+          <span className='flex-1 w-0 ml-2 truncate'>{file.name}</span>
+        </div>
+        <div className='flex-shrink-0 ml-4'>
+          <button
+            onClick={() => setIsOpen(true)}
```
Comment (theodorusclarence) [inline components/Forms/DropzoneInput.tsx:61]: type button yang ini jangan lupa riz

File: pages/form/step-2.tsx [hunk 2]
```diff
@@ -77,7 +78,13 @@ export default function StepTwoPage() {
                 <Input id='score_1' label='Score 1' />
                 <Input id='score_2' label='Score 2' />
                 <Input id='score_3' label='Score 3' />
-                <Input type='file' id='score_file' label='Score File' />
+                <DropzoneInput
+                  label='Score File'
+                  id='score_file'
+                  accept='image/png, image/jpg, image/jpeg'
+                  helperText='You can only drop .jpg, .jpeg, and .png image here'
+                  maxFiles={3}
```
Comment (theodorusclarence) [inline pages/form/step-2.tsx]: maxFiles dibikin 1 aja gapapa riz, 

biasanya kalo ada beberapa file, 1 1 inputnya wkwkwkwk,

**Review by theodorusclarence — CHANGES_REQUESTED** [review]

**Comment by theodorusclarence** [comment]
footer jangan lupa tambahin

File: components/Forms/DropzoneInput.tsx [hunk 3]
```diff
@@ -0,0 +1,201 @@
+import { MouseEvent, ReactElement, useCallback, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { FileWithPath, useDropzone } from 'react-dropzone';
+import Lightbox from 'react-image-lightbox';
+
+import { HiOutlineEye, HiOutlinePaperClip, HiX } from 'react-icons/hi';
+
+import { classNames } from '@/lib/helper';
```
Comment (theodorusclarence) [inline components/Forms/DropzoneInput.tsx]: oh sama ini, pake clsx aja riz, uda migrate gw wkwkwkwk

`import clsx from 'clsx'`

cara pakenya sama persis, ganti nama aja

File: components/Forms/DropzoneInput.tsx [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+type DropzoneInputProps = {
+  accept?: string;
+  helperText?: string;
+  id: string;
+  label: string;
+  maxFiles?: number;
+  readonly?: boolean;
```
Comment (theodorusclarence) [inline components/Forms/DropzoneInput.tsx]: sama ini, kayaknya mending pake `readOnly` biar konsisten sama yang lain

File: pages/form/step-2.tsx (hunk 2 above)
Comment (rizqitsani) [reply pages/form/step-2.tsx]: okee wkwk

File: components/Forms/DropzoneInput.tsx (hunk 4 above)
Comment (rizqitsani) [reply components/Forms/DropzoneInput.tsx]: typo :)

**Comment by theodorusclarence** [comment]
#### 📝Changed routes:


| [**/form/step-2**](https://rhf-stepform-qo3sjyu1u-theodorusclarence.vercel.app/form/step-2) | [**/**](https://rhf-stepform-qo3sjyu1u-theodorusclarence.vercel.app/) |
|:-:|:-:|
|<a href="https://rhf-stepform-qo3sjyu1u-theodorusclarence.vercel.app/form/step-2">Screenshot of /form/step-2</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/shgmXGq5ufk7AzNVykEISAbjPCTgx7dXYpDiXhVUDtFxoF_uomAPK3aEHL7YsTboUnCW_kMVwfQ33jJ83In9hg.png">(view full size)</a> |<a href="https://rhf-stepform-qo3sjyu1u-theodorusclarence.vercel.app/">Screenshot of /</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/FNTHeBh12eV5KIPToFKi14RxhLOh4SFm7qWxzA0BbSl-a_2f-7JOza0YnyQG7U26M2hBR75x4qzPOOPG_P1ung.png">(view full size)</a> |







Commit 3695424da8e71f9c1a7562471fc4007d8cef15e0 (https://rhf-stepform-qo3sjyu1u-theodorusclarence.vercel.app).

File: pages/form/step-2.tsx [hunk 5]
```diff
@@ -81,9 +81,9 @@ export default function StepTwoPage() {
                 <DropzoneInput
                   label='Score File'
                   id='score_file'
-                  accept='image/png, image/jpg, image/jpeg'
+                  accept='image/png, image/jpg, image/jpeg, application/pdf'
                   helperText='You can only drop .jpg, .jpeg, and .png image here'
```
Comment (theodorusclarence) [inline pages/form/step-2.tsx]: helper text jangan lupa wkwkwk

File: components/Forms/DropzoneInput.tsx [hunk 6]
```diff
@@ -84,15 +89,26 @@ const FilePreview = ({ deleteFile, file }: FilePreviewProps): ReactElement => {
...
-        onClick={(e) => deleteFile(e, file)}
-        className='flex mr-2 leading-none'
-      >
-        <HiX size={24} className='text-red-500 cursor-pointer' />
-      </button>
+    <li
+      key={file.name}
+      className='flex items-center justify-between py-3 pl-3 pr-4 text-sm'
+    >
       {file.name}
```
Comment (theodorusclarence) [inline components/Forms/DropzoneInput.tsx]: https://github.com/theodorusclarence/rhf-stepform/pull/1/commits/40a26e539252fd31d85b6641615a0c9083df595c#diff-c0ee6be7eeffa2e5387b467184179e3550c4ebc7351dd7e7d21ac41062d1f515R50-R56

yang ini iconsnya tetep pake aja gimana?
mungkin yang image -> pake icon HiOutlinePhotograph
sisanya pake paper clip

File: components/Forms/DropzoneInput.tsx [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+          className='flex-shrink-0 w-5 h-5 text-gray-400'
+          aria-hidden='true'
+        />
+        <span className='flex-1 w-0 ml-2 truncate'>{file.name}</span>
+      </div>
+      <div className='flex space-x-2'>
+        <UnstyledLink href={URL.createObjectURL(file)}>
+          <HiOutlineExternalLink
+            size={24}
+            className='text-gray-500 cursor-pointer hover:text-gray-700'
```
Comment (theodorusclarence) [inline components/Forms/DropzoneInput.tsx]: cursor-pointer apus ajaa

**Review by theodorusclarence — APPROVED** [review]
image

**Pull request merged**